libs/libarchfpga/src/read_common_func.h

```diff
@@ -0,0 +1,19 @@
+#ifndef READ_COMMON_FUNC_H
+#define READ_COMMON_FUNC_H
+
+#include "arch_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* function declarations */
```

Comment:
I think the common functions should be splitted from the large source files.
Also I believe this is a good time to think about what common functions we need.
My suggestion is that we need
- Check functions, such as ``check_models()`` as you have done, as well as other ``check_arch()``. If we agree, can we rename this file to ``check_arch.h``. We can also add comments to these functions so that it is easer for other developers to follow-up

Reply:
Yeah, this makes completely sense. Given the huge amount of stuff in the XML arch reader that is not strictly corresponding to parsing the file itself, I think as well that it is good to extract generic functions in separate files.

I have started to address this in https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1907/commits/e9ba680b7de5c3350183aefa0e7e8a3689b64abb and put all the check functions in an `arch_check` file.